crates/node-builder/src/builder.rs

```diff
@@ -629,6 +655,217 @@ where
     }
 }
 
+/// A [NodeBuilder] with it's launch context already configured
```

Comment:
Pretty insane trait bounds here, recommend we document them well, but ultimately OK to have complex internals if they expose powerful API